src/io/json/read/deserialize.rs

```diff
@@ -60,35 +78,44 @@ fn build_extract(data_type: &DataType) -> Extract {
 fn deserialize_boolean_into<'a, A: Borrow<Value<'a>>>(
     target: &mut MutableBooleanArray,
     rows: &[A],
-) {
+) -> Result<(), Error> {
     let iter = rows.iter().map(|row| match row.borrow() {
         Value::Bool(v) => Some(v),
         _ => None,
     });
     target.extend_trusted_len(iter);
+    Ok(())
 }
 
-fn deserialize_int_single<T>(number: Number) -> T
+fn deserialize_int_single<T>(number: Number) -> Result<T, Error>
 where
     T: NativeType + lexical_core::FromLexical + Pow10,
 {
     match number {
         Number::Float(fraction, exponent) => {
-            let integer = fraction.split(|x| *x == b'.').next().unwrap();
-            let mut integer: T = lexical_core::parse(integer).unwrap();
+            let integer = fraction.split(|x| *x == b'.').next().ok_or_else(|| {
+                Error::ExternalFormat("Failed to parse float: no bytes".to_string())
```

Comment:
Yeah, one can see it was robot - human would create a nice constructor for `Error::ExternalFormat`

Reply:
It was also kinda forced to not do that via context (it was only given this file and the above panics.md), FWIW.

I wonder if different prompting would get it to do the more elegant thing (not enough to actually bother trying this time around haha)